test/copy.js

```diff
@@ -0,0 +1,109 @@
+var assert = require('assert'),
+    rest = require('cloudcmd/lib/server/rest'),
+    path = require('path'),
+    fs   = require('fs-extra'), // use fs-extra for removeSync
+    home = process.env.HOME,
+    scratch = process.env.SCRATCH,
+    fixtures_path = path.resolve(__dirname, 'fixtures/'),
+    hometmp, scratchtmp;
+
+function s(p){
```

Comment:
We shouldn't use one character function names or variable names outside of the conventional (`n`, `i` when writing c-style loop counters, and when doing cartesian coordinates `x`, `y`, `z`).

Reply:
In this one case I have mixed feelings because the single letter dramatically increases the readability of the code once you know what the single letter does. Resorting to just ensuring that a slash is appended to all strings is relying on state which in practices is prone to error. So in the end it's a set of bad tradeoffs.

Though no one is ever going to edit this again so...

Reply:
For example:

```js
rest.copy(s(fixtures_path), s(scratchtmp), ["data"], function(){
  rest.copy(s(scratchtmp), s(target), ["data"], function(){

// vs

rest.copy(add_slash(fixtures_path), add_slash(scratchtmp), ["data"], function(){
  rest.copy(add_slash(scratchtmp), add_slash(target), ["data"], function(){
```

Reply:
Well this is actually better:

```js
rest.copy(normalize(fixtures_path), normalize(scratchtmp), ["data"], function(){
  rest.copy(normalize(scratchtmp), normalize(target), ["data"], function(){
```

though `normalize` here != `path.normalize`